### DISCUSSION TITLE
Prisma 2 delete mutation is returning null and not deleting the record

### DISCUSSION THREAD

**Discussion by gurerd2** [prisma/prisma#10902] [Q&A] [answered]
I am using ApolloServer/Prisma2/GraphQL/Typescript/MySQL/Node.js I have created two models, User and Post. My createUser and createPost mutations are working fine. However, I am having trouble getting my delete mutations working. Focusing on the deletePost here is what I have in my code.

<<schema.graphql>>

```
type Query {
    posts: [Post!]!
    users: [User!]!
  }

type Mutation {
    createPost(title: String!, body: String!): Post!
    createUser(name: String!, email: String!, password: String!): User!
    deletePost(postId: ID!): Post
    deleteAllPosts: [Post!]!
 }

type Post {
    id: ID! 
    title: String!
    body: String!
    author: User
    published: Boolean!
}

type User {
    id: ID!
    name: String!
    email: String! 
    posts: [Post!]!
}
```
<<schema.prisma>>

```
model Post {
  id                  Int      @id @default(autoincrement())
  createdAt     DateTime @default(now())
  updatedAt    DateTime @updatedAt @default(now())
  title               String
  body             String
  published     Boolean  @default(false)
  postedBy     User?    @relation(fields: [authorId], references: [id])
  authorId      Int?
}

// note that I have authorId as optional just temporarily. Another issue but not for this discussion.
  

model User {
  id        Int      @id @default(autoincrement())
  name      String
  email     String   @unique
  password  String
  posts     Post[]
}
```
<<Mutation.ts>>

```
async function deletePost(parent, { postId }, context, info) {
    return await context.prisma.post.delete(
        {
            where {
                id: parseInt(postId)
            }
        },
            info
     )
}
```
Note that the createPost and createUser are also in the Mutation.ts file and are working correctly. So I'm assuming there is no issue with the Apollo server.

When I use the GraphQL playground I use the following:

```
mutation {
  deletePost(postId: "1") {
    id
  }
}
with the following result:

{
  "data": {
    "deletePost": null
  }
}
```

I'd like the mutation to return the deleted post (at the very least the id). In addition, the database is not deleting anything. I'd appreciate any help. Thanks!

**Comment by l2D** [comment, marked as answer]
I curious, did you got `postId` in `deletePost()`? // Mutation.ts

```js
async function deletePost(parent, { postId }, context, info) {
  + console.log(`postId :>>`, postId);
    return await context.prisma.post.delete(
        {
            where {
                id: parseInt(postId)
            }
        },
            info
     )
}
```

> Note that the createPost and createUser are also in the Mutation.ts file and are working correctly. So I'm assuming there is no issue with the Apollo server.

Could you try on `updatePost()`? (I think you'll face with same issue like a `deletePost()`.)

**Reply by gurerd2** [reply]
I2D.  Thanks for your response! I have a basic newbie question. How do I see the console.log info that is printed from the deletePost function -- from the line you added in your comment above.  I tried this but I can't figure out how to see the console. It doesn't print to my terminal in Visual Studio and I tried the developer tools console in Chrome -- not there either. Where is it going?

**Reply by l2D** [reply]
Actually, It's should print in the terminal.

Note: below code I ref from [NestJS Graphql code-first](https://github.com/prisma/prisma-examples/tree/latest/typescript/graphql-nestjs)

Resolver.ts
```js
  @Mutation(() => Rating, { description: `Delete a rating` })
  async deleteRating(@Args('id') id: number) {
    const parsedId = parseInt(id.toString(), 10);
    console.log('id :>> ', id);
    console.log(`parsedId :>> `, parsedId);
    return await this.prismaService.rating.delete({
      where: { id: parsedId },
    });
  }
```
<details>
  <summary>Terminal</summary>


```bash
[10:57:09 AM] File change detected. Starting incremental compilation...

[10:57:10 AM] Found 0 errors. Watching for file changes.

[Nest] 59302  - 12/31/2021, 10:57:11 AM     LOG [NestFactory] Starting Nest application...
[Nest] 59302  - 12/31/2021, 10:57:11 AM     LOG [InstanceLoader] AppModule dependencies initialized +32ms
[Nest] 59302  - 12/31/2021, 10:57:11 AM     LOG [InstanceLoader] GraphQLSchemaBuilderModule dependencies initialized +1ms
[Nest] 59302  - 12/31/2021, 10:57:11 AM     LOG [InstanceLoader] GraphQLModule dependencies initialized +0ms
[Nest] 59302  - 12/31/2021, 10:57:11 AM     LOG [NestApplication] Nest application successfully started +133ms

🚀 Server ready at: http://localhost:3232/graphql

id :>>  3
parsedId :>>  3
```
</details>

<details>
  <summary>schema.gql</summary>
  

```gql
# ------------------------------------------------------
# THIS FILE WAS AUTOMATICALLY GENERATED (DO NOT MODIFY)
# ------------------------------------------------------
"""
`Date` type as integer. Type represents date and time as number of milliseconds from start of UNIX epoch.
"""
scalar Timestamp

type Rating {
  id: Int!
  rating: String!
  Books: [Book!]
  created_at: Timestamp!
  updated_at: Timestamp!
}

type Query {
    .
    .
    .
}

type Mutation {
  updateBook(data: BookUpdateInput!, id: String!): Book!
  deleteBook(id: String!): Book!
  deleteRating(id: Float!): Rating!
}

input BookCreateInput {
    .
    .
    .
}

input BookUpdateInput {
    .
    .
    .
}

   // Other `input`
    .
    .
    .

```
</details>


<details>
  <summary>Playground</summary>
  
![Playground - httplocalhost3232graphql_2021-12-31_11-16-12_OiS44XfO](https://user-images.githubusercontent.com/20911264/147803122-0b2ec26a-c091-4e7a-a590-292c4dc63842.png)

</details>

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
